src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -155,6 +161,14 @@ def lockfile(self) -> str:
         """
         return cast(str, self.options.experimental_lockfile)
 
+    @property
+    def is_lockfile(self) -> bool:
+        """True if these requirements represent a lockfile.
+
+        (i.e. the lockfile path is not '<none>' and class property `register_lockfile == True`.
+        """
```

Comment:
If you change the function name to `lockfile_used`, imo this docstring becomes redundant and can go. We generally prefer "self-documenting code" over comments.